pkg/operator/resourcesynccontroller/core_test.go

```diff
@@ -0,0 +1,285 @@
+			mockConfigMaps: map[string]map[string]*corev1.ConfigMap{
+				"ns1": {
+					"cm1": {
+						Data: map[string]string{
+							"ca-bundle.crt": string(validCertPEM),
+						},
+					},
+				},
+			},
+			inputLocations: []ResourceLocation{
+				{Namespace: "ns1", Name: "cm1"},
+			},
+			additionalAnnotations: certrotation.AdditionalAnnotations{
+				JiraComponent: "foobar",
+			},
+			expectModified: false,
+			expectError:    false,
+			expectedAnnotations: map[string]string{
+				annotations.OpenShiftComponent: "foobar",
+			},
+		},
+		{
+			name: "metadata modified only",
+			destinationConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cm1",
+					Namespace: "ns1",
+				},
+				Data: map[string]string{
+					"ca-bundle.crt": string(validCertPEM),
+				},
+			},
+			mockConfigMaps: map[string]map[string]*corev1.ConfigMap{
+				"ns1": {
+					"cm1": {
+						Data: map[string]string{
+							"ca-bundle.crt": string(validCertPEM),
+						},
+					},
+				},
+			},
+			inputLocations: []ResourceLocation{
+				{Namespace: "ns1", Name: "cm1"},
+			},
+			additionalAnnotations: certrotation.AdditionalAnnotations{
+				JiraComponent: "foobar",
+			},
+			expectModified: true,
+			expectError:    false,
+			expectedAnnotations: map[string]string{
+				annotations.OpenShiftComponent: "foobar",
+			},
+		},
+		{
+			name: "contents modified only",
+			destinationConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cm1",
+					Namespace: "ns1",
+					Annotations: map[string]string{
+						annotations.OpenShiftComponent: "foobar",
+					},
+				},
+				Data: map[string]string{
+					"ca-bundle.crt": string(validCertPEM),
+				},
+			},
+			mockConfigMaps: map[string]map[string]*corev1.ConfigMap{
+				"ns1": {
+					"cm1": {
+						Data: map[string]string{
+							"ca-bundle.crt": string(validCertPEM2),
+						},
+					},
+				},
+			},
+			inputLocations: []ResourceLocation{
+				{Namespace: "ns1", Name: "cm1"},
+			},
+			additionalAnnotations: certrotation.AdditionalAnnotations{
+				JiraComponent: "foobar",
+			},
+			expectModified: true,
+			expectError:    false,
+			expectedAnnotations: map[string]string{
+				annotations.OpenShiftComponent: "foobar",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			lister := &mockConfigMapLister{
+				configMaps: test.mockConfigMaps,
+			}
+
+			result, modified, err := CombineCABundleConfigMaps(test.destinationConfigMap, lister, test.additionalAnnotations, test.inputLocations...)
+
+			if test.expectError && err == nil {
+				t.Errorf("Expected error but got none")
+			}
+			if !test.expectError && err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			if test.expectModified != modified {
+				t.Errorf("Expected modified=%v but got %v", test.expectModified, modified)
+			}
+			if err == nil && result == nil {
+				t.Errorf("Expected result to not be nil when no error occurred")
+			}
+
```

Comment:
Any reason not to compare the full returned CM against an expected CM?

Reply:
Right, it would verify removal of expired certs too